patches/tModLoader/Terraria/ModLoader/UI/UICreateMod.cs

```diff
@@ -251,6 +251,7 @@ internal string GetModCsproj(string modNameTrimmed)
     <TargetFramework>net6.0</TargetFramework>
     <PlatformTarget>AnyCPU</PlatformTarget>
     <LangVersion>latest</LangVersion>
+	<ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
```

Comment:
I'll leave the config changes for jopo to review, but this would be better in `tMLMod.targets` as a separate PR. I'm not sure if this is a great idea overall, The plan has always been to actually fix the steam dll so that it's compatible with ARM, and this may ignore the warning when it's important for other mod libraries